src/robot_manager.cpp

```diff
@@ -557,6 +574,61 @@ bool RobotManager::setTargetCb(temoto_robot_manager::RobotSetTarget::Request& re
   return true;
 }
 
+
+
+bool RobotManager::getTargetCb(temoto_robot_manager::RobotGetTarget::Request& req,
+                               temoto_robot_manager::RobotGetTarget::Response& res)
+{
+  if (active_robot_->isLocal())
+  {
+    TEMOTO_INFO("Request: '%s'", req.ref_joint.c_str());
+    TEMOTO_INFO("Request: '%s'", req.respect_to.c_str());
+    res.pose = active_robot_->getTarget();
+  }
+  else
+  {
+    TEMOTO_INFO("robot is not local");
+  }
+
+  return true;
+}
+
+
+bool RobotManager::goalCb(temoto_robot_manager::RobotGoal::Request& req, temoto_robot_manager::RobotGoal::Response& res)
+{
+  active_robot_->goal("map", req.target_pose);
+  std::string act_rob_ns = active_robot_->getConfig()->getAbsRobotNamespace() + "/move_base";
+
+  //MoveBaseClient ac("/xarm7_temoto/robot_manager/robots/clearbot/move_base", true);   //For testing with clearbot
+  MoveBaseClient ac(act_rob_ns, true);  
+  
+  
+  while(!ac.waitForServer(ros::Duration(5.0))){
```

Comment:
The loop will terminate either when the server is present or the 5 second timeout is reached. I assume that the code following the while loop will not work properly, if the timeout is reached. Please add some additional safety measures to see if the server is actually there.

Reply:
No, the loop ends just when the server is present. 
Otherwise, it keeps waiting for the server to come up and it prints a message every 5 seconds.